### ISSUE TITLE
Missing XREFs to structure field references

### ISSUE THREAD

**Issue by grokeus** [Vector35/binaryninja-api#2499]
**Describe the bug**
Binary ninja doesn't show cross reference to structure fields when pointer to structure field passed to function.
for example we could take following code:
```
GLOBAL _start


section .text
_start:
        call b
        ret
b:
        mov rax, rdi
        mov qword [rax], 1;
        mov qword [rax+8], 2;
        mov qword [rax+16], 3;
        mov rbx, rax
        add rbx, 8
        mov rdi, rbx
        call c
        ret

c:
        push rbp
        mov qword [rdi], 4
        pop rbp
        ret
```

**To Reproduce**

1. load attached binary in binary ninja
2. create structure consisting of three _uint64_t_ 
3. go to function **b**
4. set argument type to pointer to **struct_1**

**Expected [struct_ref.zip](https://github.com/Vector35/binaryninja-api/files/6625595/struct_ref.zip)ehavior**
When we select field_8 of out structure we should see two XREFs

**Screenshots**
![изображение](https://user-images.githubusercontent.com/20337200/121397939-7f4fa680-c95d-11eb-9e6e-7f7336d36e3f.png)

![изображение](https://user-images.githubusercontent.com/20337200/121397956-85de1e00-c95d-11eb-873a-d68eafa383b4.png)

**Desktop (please complete the following information):**
 - OS: Windows
 - Version: 10.0.19041.867

**Additional context**
Problem not tied for x86 architecture. I have arm binary with same behaviour.

[struct_ref.zip](https://github.com/Vector35/binaryninja-api/files/6625595/struct_ref.zip)

**Comment by jeffli678** [comment]
I will investigate this ASAP

**Comment by jeffli678** [comment]
The dereference to the structure field is only created in HLIL. However, in the current implementation, the type xrefs are only collected from MLIL, which leads to the missing refs. We should also collect type xrefs from HLIL, in the same way as we do for variable xrefs

**Comment by jeffli678** [comment]
fix in dev 2.4.2931
